DBConnect.cc

```diff
@@ -1,45 +1,77 @@
-
-
 #if _AUTH_SERVER_ // defined in cmake files
 
+#include <carta-protobuf/register_viewer.pb.h>
+#include <carta-protobuf/user_preferences.pb.h>
+#include <fmt/format.h>
+#include <fmt/ostream.h>
+#include <google/protobuf/map.h>
 #include <jsoncpp/json/json.h>
 #include <jsoncpp/json/value.h>
 #include <fstream>
 #include <iostream>
 #include <sstream>
 
+#define _XOPEN_SOURCE_
 #include <bson.h>
+#include <json-c/json.h>
 #include <mongoc.h>
 #include <stdio.h>
-
+#include <unistd.h>
 #include "Carta.h"
+#include "DBConnect.h"
+#include "EventHeader.h"
+
+bool initMongoDB(mongoc_database_t** database, mongoc_client_t** client, mongoc_collection_t** collection, const char* collname) {
+    mongoc_uri_t* uri;
+    bson_error_t error;
+    std::string uri_string;
+
+    GetMongoURIString(uri_string);
+
+    mongoc_init();
+
+    uri = mongoc_uri_new_with_error(uri_string.c_str(), &error);
+    if (!uri) {
+        fmt::format(
```

Comment:
Doesn't `format` just return a string, and `print` print?

Reply:
Yep